### PR TITLE
adds a playbook and roles for vagrant deployment

### PR DESCRIPTION
**Pull request by andrewschoen**
This punts on the requirement mentioned in issue #2, but I will follow up with that and include a prod deployment playbook.

### CODE REVIEW

File: deploy/playbooks/roles/common/templates/nginx_site.conf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    ssl_certificate     /etc/ssl/certs/{{ fqdn }}-bundled.crt;
+    ssl_certificate_key /etc/ssl/private/{{ fqdn }}.key;
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+    add_header Strict-Transport-Security "max-age=31536000";
+
+    access_log  /var/log/nginx/{{ app_name }}-access.log;
+    error_log /var/log/nginx/{{ app_name }}-error.log;
+
+    # Some binaries are gigantic
+    client_max_body_size 2048m;
```
Comment (alfredodeza) [inline deploy/playbooks/roles/common/templates/nginx_site.conf]: this is no longer true

File: deploy/playbooks/roles/common/templates/systemd/shaman.service.j2 [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# {{ ansible_managed }}
+[Unit]
+Description=shaman gunicorn service
+
+[Service]
+Type=simple
+ExecStart={{ app_home }}/bin/gunicorn_pecan -w 10 -t 1200 {{ app_home }}/src/{{ app_name }}/prod.py
```
Comment (alfredodeza) [inline deploy/playbooks/roles/common/templates/systemd/shaman.service.j2]: we might want to reduce 1200 for the timeout. Maybe more like 60 (a minute!)

File: deploy/playbooks/roles/common/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+- include: systemd.yml
+  tags:
+    - systemd
+
+- include: nginx.yml
+  tags:
+    - nginx
+
+# nginx should be up and running by now, there is a bug with Ubuntu 14.04 and
```
Comment (alfredodeza) [inline deploy/playbooks/roles/common/tasks/main.yml]: this is no longer 14.04
woot!

File: deploy/playbooks/roles/common/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+- name: ensure nginx is running
+  sudo: true
+  action: service name=nginx state=started
+
+- name: ensure nginx was started and stayed up and running
+  sudo: true
+  action: service name=nginx state=started
+  register: nginx_was_running
+  when: ansible_distribution == 'Ubuntu' and ansible_distribution_release == 'trusty'
```
Comment (alfredodeza) [inline deploy/playbooks/roles/common/tasks/main.yml]: no longer running on trusty

File: deploy/playbooks/roles/common/tasks/postgresql.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+- name: restart postgresql if configuration was changed
+  service:
+    name: postgresql
+    state: restarted
+  sudo: true
+  when: pg_hba_conf.changed or
+        postgresql_conf.changed
+
+# FIXME: flag/option needed
```
Comment (alfredodeza) [inline deploy/playbooks/roles/common/tasks/postgresql.yml]: aren't we doing this all the time every time?

Comment (andrewschoen) [inline deploy/playbooks/roles/common/tasks/postgresql.yml]: Yeah, that password gets changed on every deploy.

**Pull request merged**